src/YagrCore/plugins/tooltip/utils.ts

```diff
@@ -8,3 +10,65 @@ export function escapeHTML(html: string) {
     elem.innerText = html;
     return elem.innerHTML;
 }
+
+type FindClosestAreaOptions = {
+    /** cursor X */
+    x: number;
+    /** cursor Y */
+    y: number;
+    series: AlignedData;
+};
+
+export const findClosestLines = ({x, y, series}: FindClosestAreaOptions) => {
+    const [timeline, ...areas] = series as number[][];
+
+    let i1 = 0,
```

Comment:
I believe you can find `i1` and `i2` with using `uplot.cursor.idx` or something. I think traversing whole timeline on each `setCursor` generally is a bad idea, because `tooltip.tracking` is function which can be called very frequently.